src/core/coap/coap_header.cpp

```diff
@@ -281,6 +312,56 @@ ThreadError Header::AppendContentFormatOption(MediaType aType)
     return AppendOption(coapOption);
 }
 
+ThreadError Header::AppendMaxAgeOption(uint32_t aMaxAge)
+{
+    Option coapOption;
+    uint8_t buf[4];
+
+    coapOption.mNumber = kCoapOptionMaxAge;
+
+    if (aMaxAge <= 0xff)
```

Comment:
On a second thought, are you sure this code would work? E.g. given aMaxAge = 0x00123456:
on LE machines: aMaxAge => 0x56341200, buf = { 00, 12, 34, 56 } - good order, length too long
on BE machines: aMaxAge => 0x00123456, buf = { 56, 34, 12 } - reversed order, good length.
Or am I missing something?
I think this should give the correct result:
```C++
aMaxAge = Encoding::BigEndian::HostSwap32(aMaxAge);
coapOption.mLength = 4;
coapOption.mValue = static_cast<uint8_t *>(&aMaxAge);
// skip preceding zeros, but make sure mLength is at least 1
while (coapOption.mValue[0] == 0 && coapOption.mLength > 1) {
    coapOption.mValue++;
    coapOption.mLength--;
}
```
, but not so sure if short and vague is better than long and clear.